File: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples-java/UpdateDatastore.java
import com.azure.core.util.Context;
import com.azure.resourcemanager.connectedvmware.models.Datastore;
import java.util.HashMap;
import java.util.Map;

/** Samples for Datastores Update. */
public final class Main {
    /*
     * x-ms-original-file: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples/UpdateDatastore.json
     */
    /**
     * Sample code: UpdateDatastore.
     *
     * @param manager Entry point to ConnectedVMwareManager.
     */
    public static void updateDatastore(com.azure.resourcemanager.connectedvmware.ConnectedVMwareManager manager) {
        Datastore resource =
            manager.datastores().getByResourceGroupWithResponse("testrg", "HRDatastore", Context.NONE).getValue();
        resource.update().withTags(mapOf("tag1", "value1", "tag2", "value2")).apply();
    }

    @SuppressWarnings("unchecked")
    private static <T> Map<String, T> mapOf(Object... inputs) {
        Map<String, T> map = new HashMap<>();
        for (int i = 0; i < inputs.length; i += 2) {
            String key = (String) inputs[i];
            T value = (T) inputs[i + 1];
            map.put(key, value);
        }
        return map;
    }
}

File: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples-js/UpdateDatastore.js
const { AzureArcVMwareManagementServiceAPI } = require("@azure/arm-connectedvmware");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to API to update certain properties of the datastore resource.
 *
 * @summary API to update certain properties of the datastore resource.
 * x-ms-original-file: specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-01-10-preview/examples/UpdateDatastore.json
 */
async function updateDatastore() {
  const subscriptionId = "fd3c3665-1729-4b7b-9a38-238e83b0f98b";
  const resourceGroupName = "testrg";
  const datastoreName = "HRDatastore";
  const body = { tags: { tag1: "value1", tag2: "value2" } };
  const options = { body };
  const credential = new DefaultAzureCredential();
  const client = new AzureArcVMwareManagementServiceAPI(credential, subscriptionId);
  const result = await client.datastores.update(resourceGroupName, datastoreName, options);
  console.log(result);
}

updateDatastore().catch(console.error);
